runtime/jcl/common/mgmtmemory.cpp

```diff
@@ -132,13 +86,13 @@ Java_com_ibm_java_lang_management_internal_MemoryMXBeanImpl_getNonHeapMemoryUsag
 		UDATA *ramClassFreqUDATABlockFreeListPtr = frequentlyAccessedBlockPtr->ramClassUDATABlockFreeList;
 		UDATA *ramClassInFreqUDATABlockFreeListPtr = inFrequentlyAccessedBlockPtr->ramClassUDATABlockFreeList;
 
-		subtractUDATABlockChain(ramClassSub4gUDATABlockFreeListPtr, &used);
-		subtractUDATABlockChain(ramClassFreqUDATABlockFreeListPtr, &used);
-		subtractUDATABlockChain(ramClassInFreqUDATABlockFreeListPtr, &used);
+		VM_VMHelpers::subtractUDATABlockChain(ramClassSub4gUDATABlockFreeListPtr, (U_64 *)&used);
+		VM_VMHelpers::subtractUDATABlockChain(ramClassFreqUDATABlockFreeListPtr, (U_64 *)&used);
+		VM_VMHelpers::subtractUDATABlockChain(ramClassInFreqUDATABlockFreeListPtr, (U_64 *)&used);
 
-		subtractFreeListBlocks(sub4gBlockPtr, &used);
-		subtractFreeListBlocks(frequentlyAccessedBlockPtr, &used);
-		subtractFreeListBlocks(inFrequentlyAccessedBlockPtr, &used);
+		VM_VMHelpers::subtractFreeListBlocks(sub4gBlockPtr, (U_64 *)&used);
+		VM_VMHelpers::subtractFreeListBlocks(frequentlyAccessedBlockPtr, (U_64 *)&used);
+		VM_VMHelpers::subtractFreeListBlocks(inFrequentlyAccessedBlockPtr, (U_64 *)&used);
```

Comment:
There’s a signedness mismatch between `jlong` (signed) and `U_64` (unsigned). `jlong` aligns with `I_64`, not `U_64`. Since the helpers subtract from used, the accumulator should be signed or we risk unsigned underflow and confusing wrap-around. Can we change `chunkSize` and `blockSize` to `I_64` in `ClassLoaderStatisticsEntry`?

Reply:
Changed `blockSize` to `I_64` and added a check in `writeClassLoaderStatisticsEvent` to avoid writing a negative `blockSize`.

Reply:
I think it's fine to leave it as `U_64`?